pkg/sources/filesystem/filesystem.go

```diff
@@ -151,7 +151,13 @@ func (s *Source) scanFile(ctx context.Context, path string, chunksChan chan *sou
 	defer inputFile.Close()
 	logger.V(3).Info("scanning file")
 
-	reReader, err := diskbufferreader.New(inputFile)
```

Comment:
If the file has a .deb or .rpm extension, but is not actually a file of that type, we'll end up running through the reader, but not be able to reset it.